### PR TITLE
Secure Signals GMA DevApp

### PR DESCRIPTION
**Pull request by bleege**
Creates a DevApp for the purposes of:

* Providing a test harness for UID2 to Google GMA integration via the Secure Signals GMA Plugin.

Related Work:
* Ported Google's IMA BasicExample app for germane app example
    * https://github.com/googleads/googleads-mobile-android-examples/tree/main/java/admob/BannerExample

iOS Version:
* https://github.com/IABTechLab/uid2-ios-plugin-google-gma/pull/2

### CODE REVIEW

File: securesignals-gma-devapp/src/main/java/com/uid2/dev/MyActivity.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+
+/**
+ * Port of <a href="https://github.com/googleads/googleads-mobile-android-examples/tree/main/java/admob/BannerExample">BannerExample</a>
+ */
+public class MyActivity extends AppCompatActivity {
```
Comment (IanDBird) [inline securesignals-gma-devapp/src/main/java/com/uid2/dev/MyActivity.java:31]: Similar to the IMA review, there are a number of issues in this file, including:
 - `UID2Manager` should be initialised in an `Application` class, not an Activity
 - Line lengths of comments should use full 120
 - Deserialisation of initial identity could be improved (as in, using the existing `fromJson` method - once made static)
 - Name of activity layout should be `main_activity`.
 - etc

File: securesignals-gma-devapp/src/main/res/drawable-v24/ic_launcher_foreground.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (IanDBird) [inline securesignals-gma-devapp/src/main/res/drawable-v24/ic_launcher_foreground.xml:1]: XML based drawables shouldn't be required for a dev app.

File: securesignals-gma-devapp/src/main/res/mipmap-anydpi-v26/ic_launcher.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (IanDBird) [inline securesignals-gma-devapp/src/main/res/mipmap-anydpi-v26/ic_launcher.xml:2]: Lets remove this and the associated xml drawables.

File: securesignals-gma-devapp/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (IanDBird) [inline securesignals-gma-devapp/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml:2]: Lets remove this and the associated xml drawables.

File: securesignals-gma-devapp/src/main/res/values/styles.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<resources>
```
Comment (IanDBird) [inline securesignals-gma-devapp/src/main/res/values/styles.xml:1]: Nit: Style shouldn't be required.

**Review by IanDBird — CHANGES_REQUESTED** [review]
There's a collection of nits that we should ideally update. Also, as mentioned in the other review, for consistency it should be in a module called `dev-app`, not `devapp`.

File: securesignals-gma-devapp/.gitignore [hunk 6]
```diff
@@ -0,0 +1 @@
+/build
```
[the remark this answers is not in the document]
Comment (bleege) [reply securesignals-gma-devapp/.gitignore:1]: Done.

File: securesignals-gma-devapp/build.gradle [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+    implementation project(path: ':securesignals-gma')
+    compileOnly 'com.uid2:uid2-android-sdk:0.1.0'
+    compileOnly 'com.google.android.gms:play-services-ads:22.0.0'
+
+    implementation 'androidx.multidex:multidex:2.0.1'
+    implementation 'androidx.appcompat:appcompat:1.6.1'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+
+    testImplementation 'junit:junit:4.13.2'
```
[the remark this answers is not in the document]
Comment (bleege) [reply securesignals-gma-devapp/build.gradle:40]: Done.

File: securesignals-gma-devapp/src/main/AndroidManifest.xml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+            android:name="com.google.android.gms.ads.APPLICATION_ID"
+            android:value="ca-app-pub-3940256099942544~3347511713"/>
+
+        <activity
+            android:exported="true"
+            android:name="com.uid2.dev.MyActivity"
+            android:label="@string/app_name" >
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
```
[the remark this answers is not in the document]
Comment (bleege) [reply securesignals-gma-devapp/src/main/AndroidManifest.xml:21]: Done.

File: securesignals-gma-devapp/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <application
```
[the remark this answers is not in the document]
Comment (bleege) [reply securesignals-gma-devapp/src/main/AndroidManifest.xml:4]: Done.

File: securesignals-gma-devapp/src/main/java/com/uid2/dev/MyActivity.java (hunk 1 above)
Comment (bleege) [reply securesignals-gma-devapp/src/main/java/com/uid2/dev/MyActivity.java:31]: * `UID2Manager` initialization moved to `GMADevApplication`.
* Changed `MyActivity` to `BannerActivity` to differentiate between DevApps.  Primary DevApp already uses `MainActivity`

File: securesignals-gma-devapp/src/main/res/mipmap-anydpi-v26/ic_launcher.xml (hunk 3 above)
Comment (bleege) [reply securesignals-gma-devapp/src/main/res/mipmap-anydpi-v26/ic_launcher.xml:2]: Done.

File: securesignals-gma-devapp/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml (hunk 4 above)
Comment (bleege) [reply securesignals-gma-devapp/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml:2]: Done.

File: securesignals-gma-devapp/src/main/res/values/styles.xml (hunk 5 above)
Comment (bleege) [reply securesignals-gma-devapp/src/main/res/values/styles.xml:1]: Android disagrees when I removed it:

Screenshot 2023-04-19 at 11 36 00 AM

**Comment by bleege** [comment]
The requested changes have now been made.  PR is ready for 👀 again.

**Review by IanDBird — APPROVED** [review]
Looks good!

**Comment by bleege** [comment]
I've rebased on to `main` now that #29 has merged and caused a small conflict with `settings.gradle`.

**Review by IanDBird — APPROVED** [review]

**Pull request merged**